File: exercises/quarkus/payment-service/src/main/java/com/example/payment/domain/model/AuthorizationId.java
package com.example.payment.domain.model;

import java.util.Objects;
import java.util.UUID;

public record AuthorizationId(String value) {

    private static final String PREFIX = "auth_";

    public AuthorizationId {
        Objects.requireNonNull(value, "AuthorizationId value");
        if (value.isBlank()) {
            throw new IllegalArgumentException("AuthorizationId value must not be blank");
        }
    }

    public static AuthorizationId generate() {
        return new AuthorizationId(PREFIX + UUID.randomUUID());
    }

    @Override
    public String toString() {
        return value;
    }
}

File: exercises/dotnet/PaymentService/Domain/Models/AuthorizationId.cs
namespace PaymentService.Domain.Models;

/// <summary>
/// Value object wrapping a payment authorization identifier.
/// </summary>
public record AuthorizationId(string Value)
{
    private const string Prefix = "auth_";

    public static AuthorizationId Generate() => new($"{Prefix}{Guid.NewGuid()}");

    public override string ToString() => Value;
}
